src/Microsoft.AspNet.Razor.Runtime/Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -224,18 +228,24 @@ public void AddTagHelperAttribute(string name, object value)
         }
 
         /// <summary>
-        /// Execute and retrieve the rendered child content asynchronously.
+        /// Execute children asynchronously with the given <paramref name="encoder"/> in scope and return their
+        /// rendered content.
         /// </summary>
+        /// <param name="useCachedResult">If <c>true</c> multiple calls to this method will not cause re-execution
+        /// of child content; cached content will be returned.</param>
+        /// <param name="encoder">
+        /// The <see cref="HtmlEncoder"/> to use. Does not override all HTML encodings which may occur.
```

Comment:
FYI @DamianEdwards touched on the following bit when we talked offline.
> Though changing this `HtmlEncoder` has little affect on other pages or views, it does affect all nested tag helpers.

Main change we'll make is to disable memoization of the `GetChildContentAsync()` call by default if user calls the `GetChildContentAsync(HtmlEncoder)` overload.

I will go a bit further and pass `useCachedResult: encoder != null` to the full overload in that method. That way `GetChildContentAsync(encoder: null)` will behave :100:% identically to `GetChildContentAsync()`.